core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -178,6 +179,7 @@ public class JGroupsTransport implements Transport, ChannelListener, AddressGene
    protected ExecutorService nonBlockingExecutor;
    @Inject protected CacheManagerJmxRegistration jmxRegistration;
    @Inject protected JGroupsMetricsManager metricsManager;
+   @Inject protected JGroupsRequestTracing requestTracing;
```

Comment:
I do not like this at all tbh. All you did was move the code to another class but it still exists in all code paths (happy or not). I would still suggest using a request factory component with two implementations, one for when global tracing is enabled and another when it is not. The one for when global tracing is disabled will just return the new SingleTargetRequest/MultiTargetRequest etc and the one when tracing is enabled would delegate to the disabled one but register with the operation like you are doing for the persistence manager.

This way jgroups will have no side effect to performance when global tracing is disabled.

Reply:
I tried to implement the idea of having a factory to produce SingleTargetRequest/MultiTargetRequest. But instead of touching it for add tracing I tried following a solution similar to the request log.

Reply:
You don't need to change the `*Request` classes at all. See my example here: https://github.com/pruivo/infinispan/commit/528a9e5c24ee6fd1514225ce71bebcf1e479925c#diff-1278c09490b3d1cb12ba57e301d6eb0e565ae1ed32a8064ea9e23ef344d8ac2fR1203

Reply:
I integrated the @pruivo great idea here ;)